scripts: dts: gen_legacy_defines: mark DT_INST macros deprecated

galak opened the pull request:
Add a __WARN("Macro is deprecated") to all DT_INST macros now that all
in tree users should have been converted to the new macros.

This is intended to make sure any PRs don't introduce new usages of
these macros.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
scripts: dts: gen_legacy_defines: mark DT_INST macros deprecated

Add a __WARN("Macro is deprecated") to all DT_INST macros now that all
in tree users should have been converted to the new macros.

This is intended to make sure any PRs don't introduce new usages of
these macros.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
@mbolivar-nordic @carlescufi my intent is that we'd revert this before the release.

carlescufi reviewed (approved):
LGTM but would like @mbolivar-nordic to approve

mbolivar-nordic commented on scripts/dts/gen_legacy_defines.py:
```diff
@@ -746,6 +746,9 @@ def out(ident, val, aliases=(), deprecation_msg=None):
     for alias in aliases:
         if alias != ident:
+            if alias.startswith("INST_"):
+                deprecation_msg = "Macro is deprecated"
```
Are you sure nobody is using these with `#ifdef`? As I mentioned in https://github.com/zephyrproject-rtos/zephyr/pull/24400, that won't work.

galak replied:
We'll from my grep it showed no one obvious.  I figured just by introducing the PR we'll find some cases (and have).

After merged, this will get a nightly run so we can see if there are any other cases that got missed.

mbolivar-nordic replied:
OK

mbolivar-nordic reviewed (approved):
May cause some issues, but removing by code freeze means they're affecting bleeding edge users.

galak committed:
scripts: dts: gen_legacy_defines: mark DT_*_BUS_<BUS> macros deprecated

Add a __WARN("Macro is deprecated") to all DT_<COMPAT>_BUS_<BUS> macros
now that all in tree users should have been converted to the new macros.

This is intended to make sure any PRs don't introduce new usages of
these macros.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>
